dev guide release post, 0.7.0 (#291)
ropensci/roweb3: content/blog/2021-11-18-devguide-0.7.0/index.md

Chat-ops automation has been critical to JOSS’s success, and shows huge promise for automating even more tedious tasks like frequent software checks and compliance with best practices.

If you are curious about this topic, do not miss our next [community call: Enhancing Software Peer Review with GitHub Automation](/dec2021-automation/)!
As usual resources and a close-captioned recording will be posted afterwards.

## Still some work for editors 😉

Fear not, software peer review volunteer [editors](https://ropensci.org/software-review/) still participate a lot in the process.
We've formalized [editor responsabilities](https://devdevguide.netlify.app/editorguide.html#editors-responsabilities), in particular outlining editor etiquette around submissions handled by other editors.

> You do not have to keep track of other submissions, but if you do notice an issue with a package that is being handled by another editor, feel free to raise that issue directly with the other editor, or post the concern to editors-only channel on slack. 

> Thanks to the new statistical software peer-review editor [Toby Dylan Hocking](https://github.com/tdhock) we've made it explicit that editors have to look for __two__ reviewers.

Reviewer: Ping @tdhock here, by way of gratitude for his assistance :+1:

Author: Indeed, thanks again!

We've sprinkled the dev guide with mentions of rOpenSci projects relevant to software review participants:

* We've linked to Statistical Software Peer Review from the [software review introduction](https://devdevguide.netlify.app/softwarereviewintro.html) and from the [very first page of the book](https://devdevguide.netlify.app/index.html),

* We present the [pkgcheck](https://docs.ropensci.org/pkgcheck/) package created and maintained by Mark Padgham in the author guide as using it enables authors to ensure their package is ready for submission.

* In the packaging guide, we now give precise examples of [tools for installation instructions](https://devdevguide.netlify.app/building.html#readme): the r-lib remotes and pak packages but also [R-universe](/r-universe/)!

## Setting up your package to foster a community

Following our community call ["Set Up Your Package to Foster a Community"](/commcalls/apr2021-pkg-community/) that has been [summarized by  Sina Rüeger](/blog/2021/04/28/commcall-pkg-community/), we've added guidance to our [collaboration guide](https://devdevguide.netlify.app/collaboration.html).

## Updates to reviewer guidance

> The work on pkgcheck and buffy means we were able to remove some elements from reviewers' guidance: 

Author: I've opened an issue in the blog guide https://github.com/ropensci-org/blog-guidance/issues/184

* We no longer ask reviewers to run covr but we ask them to pay attentions to skipped tests.

* We no longer ask reviewers to run goodpractice as goodpractice is run by the pkgcheck package whose checks are triggered on the cloud by the bot.

We are thrilled to remove some workload from reviewers as we are aware that reviewing a package is a lot of work. 
Thanks to all reviewers for their incredible contributions to Software Peer Review!

## Miscellaneous guidance updates

The requirement to have a vignette is now more precise, there has to be at least one _HTML_ vignette. 
That requirement is part of pkgcheck checks.

> Thanks to an idea of editor [Anna Krystalli](https://github.com/annakrystalli), the dev guide now includes an explicit [subsection about version control](https://devdevguide.netlify.app/building.html#version-control), and pkgcheck's checks include looking for "scrap files" such as `.DS_Store` files, that should be gitignored.

Reviewer 2: ```suggestion
Thanks to an idea of editor [Anna Krystalli](https://github.com/annakrystalli), the dev guide now includes an explicit [subsection about version control](https://devdevguide.netlify.app/building.html#version-control), and `pkgcheck`'s checks include looking for "scrap files" such as `.DS_Store` files, that should be `.gitignore`'d.
```

---

> title: "rOpenSci Dev Guide 0.6.0: Updates"

Reviewer 3: :grin: 
```suggestion
title: "rOpenSci Dev Guide 0.7.0: Updates"
```

We've sprinkled the dev guide with mentions of rOpenSci projects relevant to software review participants:

* We've linked to Statistical Software Peer Review from the [software review introduction](https://devdevguide.netlify.app/softwarereviewintro.html) and from the [very first page of the book](https://devdevguide.netlify.app/index.html),

* We present the [pkgcheck](https://docs.ropensci.org/pkgcheck/) package created and maintained by Mark Padgham in the author guide as using it enables authors to ensure their package is ready for submission.

* In the packaging guide, we now give precise examples of [tools for installation instructions](https://devdevguide.netlify.app/building.html#readme): the r-lib remotes and pak packages but also [R-universe](/r-universe/)!

## Setting up your package to foster a community

Following our community call ["Set Up Your Package to Foster a Community"](/commcalls/apr2021-pkg-community/) that has been [summarized by  Sina Rüeger](/blog/2021/04/28/commcall-pkg-community/), we've added guidance to our [collaboration guide](https://devdevguide.netlify.app/collaboration.html).

## Updates to reviewer guidance

> The work on pkgcheck and buffy means we were able to remove some elements from reviewers' guidance: 

Reviewer 3: What is buffy? Perhaps add a footnote? You talk about pkgcheck above, so I think that's clear.

Author: I rephrased, good point!

* We no longer ask reviewers to run goodpractice as goodpractice is run by the pkgcheck package whose checks are triggered on the cloud by the bot.

We are thrilled to remove some workload from reviewers as we are aware that reviewing a package is a lot of work. 
Thanks to all reviewers for their incredible contributions to Software Peer Review!

## Miscellaneous guidance updates

The requirement to have a vignette is now more precise, there has to be at least one _HTML_ vignette. 
That requirement is part of pkgcheck checks.

Thanks to an idea of editor [Anna Krystalli](https://github.com/annakrystalli), the dev guide now includes an explicit [subsection about version control](https://devdevguide.netlify.app/building.html#version-control), and pkgcheck's checks include looking for "scrap files" such as `.DS_Store` files, that should be `.gitignore`'d.

We've added some more guidance on [making packages easier to cite](https://devdevguide.netlify.app/building.html#version-control) following our [blog post on the same topic](/blog/2021/02/16/package-citation/).

## Policy update: scope clarification for data extraction and munging

> We've clarified the definition of the [categories](https://devdevguide.netlify.app/policies.html#package-categories) " data extraction" and "data munging" by adding examples.

Reviewer 3: ```suggestion
We've clarified the definition of the [categories](https://devdevguide.netlify.app/policies.html#package-categories) "data extraction" and "data munging" by adding examples.
```
